daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/FatContractInstance.scala

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.lf
+package transaction
+
+import data.{Bytes, Ref, Time}
+import value.{CidContainer, Value}
+
+// Should replaces value.ContractInstance in the whole daml/canton codespace
+// TODO: Rename ContractInstance once value.ContractInstance is properly deprecated
+final case class FatContractInstance private (
+    version: TransactionVersion,
+    contractId: Value.ContractId,
+    templateId: Ref.TypeConName,
+    arg: Value,
+    keyOpt: Option[GlobalKey],
+    maintainers: Set[Ref.Party],
+    nonMaintainerSignatories: Set[Ref.Party],
+    nonSignatoryStakeholders: Set[Ref.Party],
+    createTime: Time.Timestamp,
+    salt: Bytes = Bytes.Empty,
```

Comment:
I suggest we are complelty explicit about driver meta data, not 100% though. 
If the salt is 32 bytes long I call also uses `crypto.Hash`.

Reply:
Canton salts are not 32 bytes long.